### PR TITLE
Fixed Error using TargetListenerID with AzApplicationGatewayRedirectConfiguration cmdlets (Issue 8756)

### PR DESCRIPTION
**Pull request by preardon**
using TargetListner.ID instead of TargerListner.Id

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by markcowl** [comment]
@MikhailTryakhov Can you please have someone from the network team take a look at this issue?  It would also be nice for Network to add some regression protection for this scenario

**Comment by MiYanni** [comment]
@aksgupta @number213 Please review when you have the chance.

File: src/Network/Network/ApplicationGateway/RedirectConfiguration/AzureApplicationGatewayRedirectConfigurationBase.cs [hunk 1]
```diff
@@ -93,7 +93,7 @@ public PSApplicationGatewayRedirectConfiguration NewObject()
             if (!string.IsNullOrEmpty(this.TargetListenerID))
             {
                 redirectConfiguration.TargetListener = new PSResourceId();
-                redirectConfiguration.TargetListener.Id = this.TargetListener.Id;
```
Comment (akshaysngupta) [inline src/Network/Network/ApplicationGateway/RedirectConfiguration/AzureApplicationGatewayRedirectConfigurationBase.cs]: Please add this as well.
```powershell
if (this.TargetListener != null)
{
    this.TargetListenerID = this.TargetListener.Id;
}

if (!string.IsNullOrEmpty(this.TargetListenerID))
{
    redirectConfiguration.TargetListener = new PSResourceId();
    redirectConfiguration.TargetListener.Id = this.TargetListener.Id;
    redirectConfiguration.TargetListener.Id = this.TargetListenerID;
}
```

Comment (preardon) [reply src/Network/Network/ApplicationGateway/RedirectConfiguration/AzureApplicationGatewayRedirectConfigurationBase.cs]: I've added this in commit 601bc6a

**Pull request merged**